plugins/com.google.cloud.tools.eclipse.util/src/com/google/cloud/tools/eclipse/util/DependencyResolver.java

```diff
@@ -58,49 +58,57 @@
    */
   public static Collection<Artifact> getTransitiveDependencies(
       String groupId, String artifactId, String version, IProgressMonitor monitor)
-          throws CoreException {
-
-    ICallable<List<Artifact>> callable = new ICallable<List<Artifact>>() {
-      @Override
-      public List<Artifact> call(IMavenExecutionContext context, IProgressMonitor monitor)
-          throws CoreException {
-        DependencyFilter filter = DependencyFilterUtils.classpathFilter(JavaScopes.RUNTIME);
-        // todo we'd prefer not to depend on m2e here
-        RepositorySystem system = MavenPluginActivator.getDefault().getRepositorySystem();
-
-        CollectRequest collectRequest = new CollectRequest();
-        Artifact artifact = new DefaultArtifact(groupId + ":" + artifactId + ":" + version);
-
-        collectRequest.setRoot(new Dependency(artifact, JavaScopes.RUNTIME));
-        collectRequest.setRepositories(centralRepository(system));
-        DependencyRequest request = new DependencyRequest(collectRequest, filter);
-
-        // ensure checksum errors result in failure
-        DefaultRepositorySystemSession session =
-            new DefaultRepositorySystemSession(context.getRepositorySession());
-        session.setChecksumPolicy(RepositoryPolicy.CHECKSUM_POLICY_FAIL);
-
-        try {
-          List<ArtifactResult> artifacts =
-              system.resolveDependencies(session, request).getArtifactResults();
-          List<Artifact> dependencies = new ArrayList<>();
-          for (ArtifactResult result : artifacts) {
-            Artifact dependency = result.getArtifact();
-            dependencies.add(dependency);
+      throws CoreException {
+
+    ICallable<List<Artifact>> callable =
+        new ICallable<List<Artifact>>() {
+          @Override
+          public List<Artifact> call(IMavenExecutionContext context, IProgressMonitor monitor)
```

Comment:
should this monitor be converted to a submonitor here? or perhaps before calling execute?

Reply:
We pass the monitor into our `MavenUtils#runWithRule()`, which does convert it into a SubMonitor.  This particular ICallable is defined by m2e.

Reply:
I'll try refactoring this callable-within-a-callable but it's tricky and will need some more time.